File: archetype-java/src/main/resources/archetype-resources/src/main/java/health/HelloHealthCheck.java
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.health;

import com.yammer.metrics.core.HealthCheck;
import org.springframework.stereotype.Service;

import java.util.Date;

@Service
public class HelloHealthCheck extends HealthCheck {

    public HelloHealthCheck() {
        super("hello");
    }

    @Override
    protected Result check() throws Exception {
        System.out.println(new Date() +" works");
        return Result.healthy("The Date class works!");
    }
}

File: archetype-scala/src/main/resources/archetype-resources/src/main/scala/health/HelloHealthCheck.scala
#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.health

import org.springframework.stereotype.Service
import com.yammer.metrics.core.HealthCheck
import java.util.Date
import com.yammer.metrics.core.HealthCheck.Result

@Service
class HelloHealthCheck extends HealthCheck("hello") {
  def check() = {
    println(new Date()+ " works")
    Result.healthy("The Date class works")
  }
}
